pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/impl/SimpleLoadManagerImpl.java

```diff
@@ -341,88 +281,69 @@ public void start() throws PulsarServerException {
                     LOADBALANCER_DYNAMIC_SETTING_LOAD_FACTOR_MEM_ZPATH, SETTING_NAME_LOAD_FACTOR_MEM,
                     this.realtimeMemoryLoadFactor);
         } catch (Exception e) {
-            log.error("Unable to create znode - [{}] for load balance on zookeeper ", brokerZnodePath, e);
+            log.error("Unable to create node - [{}] for load balance on metadata store", brokerLockPath, e);
             throw new PulsarServerException(e);
         }
     }
 
     @Override
     public void disableBroker() throws Exception {
-        if (isNotEmpty(brokerZnodePath)) {
-            pulsar.getZkClient().delete(brokerZnodePath, -1);
+        if (brokerLock != null) {
+            brokerLock.release().join();
         }
     }
 
-    @Override
-    public Deserializer<LoadReport> getLoadReportDeserializer() {
-        return loadReportDeserializer;
-    }
-
-    public ZooKeeperChildrenCache getActiveBrokersCache() {
-        return this.availableActiveBrokers;
-    }
-
     @Override
     public Set<String> getAvailableBrokers() throws Exception {
-        return this.availableActiveBrokers.get();
-    }
-
-    public ZooKeeperDataCache<LoadReport> getLoadReportCache() {
-        return this.loadReportCacheZk;
+        return new HashSet<>(loadReports.listLocks(LOADBALANCE_BROKERS_ROOT).join());
```

Comment:
I am seeing this "listLocks(SOMETHING).join" quite often, perhaps we could add a specific method in order to not need to write always this complicated statement.
Also it is common to require a "copy", like in this case and in other points

Reply:
Simplified these

Reply:
It's not for the copy, rather to convert into a `Set`